src/main/java/eu/dissco/core/translator/repository/DwcaRepository.java

```diff
@@ -28,48 +37,112 @@
 @RequiredArgsConstructor
 public class DwcaRepository {
 
-  private final Field<String> idField = DSL.field("dwcaid", String.class);
+  private final Field<String> dwcaIDField = DSL.field("dwcaid", String.class);
+  private final Field<String> uniqueIDField = DSL.field("id", String.class);
   private final Field<JSONB> dataField = DSL.field("data", JSONB.class);
 
   private final ObjectMapper mapper;
   private final DSLContext context;
-  private final BatchInserter batchInserter;
+  private final DataMappingComponent dataMappingComponent;
 
-  public void createTable(String tableName) {
+  private static String getAccessURI(Pair<String, JsonNode> dbRecord, String uniqueField) {
+    var fieldsToCheck = new ArrayList<>(AccessURI.DWCA_TERMS);
```

Comment:
If the only difference between these 2 functions is the list of fields to check, can you combine them and make it a parameter?

Reply:
:+1: